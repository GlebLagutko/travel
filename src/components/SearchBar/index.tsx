import React from 'react';
import {useSelector} from 'react-redux'

const languageState = state => state.value.language;

const SearchBar = ({setKeyword}) => {

    const language = useSelector(languageState);

    const BarStyling = {width: "20rem", background: "#F2F1F9", border: "none", padding: "0.5rem"};
    return (
        <input
            style={BarStyling}
            key="random1"
            placeholder={language === "en" ? "Search" : language === "ru" ? "Поиск" : "Suche"}
            onChange={(event) => setKeyword(event.currentTarget.value)}
        />
    );
}

export default SearchBar
