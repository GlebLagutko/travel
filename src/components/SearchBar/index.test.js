import {create} from 'react-test-renderer';

import React from "react";
import SearchBar from ".";
import configureStore from 'redux-mock-store'
import {Provider} from 'react-redux'

describe('SearchBar', () => {
  it('renders correctly', async () => {

    const initialState = {value: {language: "en"}};
    const mockStore = configureStore();
    let store = mockStore(initialState);

    const testRenderer = create(
      <Provider store={store}><SearchBar/></Provider>
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();

  });

});