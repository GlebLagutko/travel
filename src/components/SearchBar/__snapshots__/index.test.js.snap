// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<input
  onChange={[Function]}
  placeholder="Search"
  style={
    Object {
      "background": "#F2F1F9",
      "border": "none",
      "padding": "0.5rem",
      "width": "20rem",
    }
  }
/>
`;
